### DISCUSSION TITLE
containsPoint, perPixelTargetFind and Eraser

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7499] [General]
Now that eraser is a part of fabric I have found an edge case (during the development of `Layer`) that makes me think about `perPixelTargetFind` and `containsPoint`.
If an object is partially erased and now we what to use `perPixelTargetFind` we are in a situation that is not well defined.
`perPixelTargetFind` checks if the hit point is transparent, returning accordingly.
This is a problem because I understand `perPixelTargetFind` as really clicking the object and not it's bounding box like the [ladybug demo](http://fabricjs.com/per-pixel-drag-drop).
Does clicking the erased part of an object count as really clicking an object? I would argue that in different context the answer varies.

**Should we add a flag to run `perPixelTargetFind` without eraser?**

This is related to #7473.
Nested selection requires some logic handling `perPixelTargetFind` or checking if nested objects are inside the clipped area of their parent. 

@asturur

**Comment by asturur** [comment]
So in theory the perPixelTargetFind inspects the objectCache when available.
It first check if is inside the bounding box, if it is, then it check for pixels.

**Comment by ShaMan123** [comment]
So should we add another step that checks if the object has an eraser and if so if the pixel check returns true we check if the point is contained within a path of the eraser?
